api/src/zero/api/handlers.clj

```diff
@@ -127,20 +128,25 @@
               (jdbc/query tx)
               first unnilify succeed))))))
 
-(defn get-workload
+(defn get-workload!
   "List all workloads or the workload with UUID in REQUEST."
   [request]
   (zero.api.handlers/print-handler-error
    (jdbc/with-db-transaction [tx (postgres/zero-db-config)]
-     (->> (if-let [uuid (get-in request [:parameters :query :uuid])]
-            (do
-              (logr/infof "get-workload endpoint called: uuid=%s" uuid)
-              [{:uuid uuid}])
-            (do
-              (logr/info "get-workload endpoint called with no uuid, querying all")
-              (jdbc/query tx ["SELECT uuid FROM workload"])))
-          (mapv (partial postgres/get-workload-for-uuid tx))
-          succeed))))
+     (let [uuid-vec (->> (if-let [uuid (get-in request [:parameters :query :uuid])]
+                           (do
+                             (logr/infof "get-workload endpoint called: uuid=%s" uuid)
+                             [{:uuid uuid}])
+                           (do
+                             (logr/info "get-workload endpoint called with no uuid, querying all")
+                             (jdbc/query tx ["SELECT uuid FROM workload"])))
+                      (mapv (partial postgres/get-workload-for-uuid tx)))]
+       ;;sync the workflow statuses with cromwell in another thread
+       (future
+         (jdbc/with-db-transaction [another-tx (postgres/zero-db-config)]
+         (util/do-or-nil (postgres/update-workload-for-uuid! another-tx uuid-vec))))
```

Comment:
not sure if this is anti-pattern 🤔  also not sure if this is "clojure-diomatic"

Reply:
I'm not sure why you've moved the update into a future.... The whole point of this is to get a snapshot at the point of call of the `workload` endpoint.

Reply:
> The whole point of this is to get a snapshot at the point of call of the workload endpoint.

OK, I thought this "snapshot" is a snapshot of the current DB, rather than the snapshot of the true statuses in Cromwell. The use of `future` is purely based on the observation that the update process took ~7min to finish, which would absolutely fail the request if this was synchronous.